pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,18 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    export interface HttpOptions {
+        address: string;
+        port: number;
+        headers: { [key: string]: string };
+      }
+    
+      export interface HttpInstance {
+        request(options: { path: string; method: string; body: string }): Promise<string>;
+        get(path: string): Promise<string>;
+        post(path: string, body?: any): Promise<string>;
+      }
+    
+      export function http(options: HttpOptions): HttpInstance;
```

Comment:
The more common way to cal this is with `(endpoint)`, and `(endpoint, {options})` is also allowed. See https://cockpit-project.org/guide/latest/cockpit-http.html#cockpit-http-constructor